src/cargo/util/context/mod.rs

```diff
@@ -158,6 +158,29 @@ macro_rules! get_value_typed {
     };
 }
 
+pub const TOP_LEVEL_CONFIG_KEYS: &[&str] = &[
```

Comment:
How do we want to maintain this list? I guess it'll just be best-effort?

Reply:
Yeah.  This is why I had it in this file to make it easier to keep it up to date.  If we had a schema file, we could read the schema file in and generate it.